azkaban-common/src/main/java/azkaban/executor/container/watch/AzPodStatusExtractor.java

```diff
@@ -568,33 +569,42 @@ private void logAppContainerStatuses() {
    * @return
    */
   public AzPodStatus createAzPodStatus() {
+    // All pod statuses are checked for any status extracted from the pod event. If multiple
+    // statuses are detected, the first derived pod status will be returned, and the conflict
+    // will be logged
+    List<AzPodStatus> derivedStatuses = new ArrayList<>();
+
     if (checkForAzPodRequested()) {
-      return AzPodStatus.AZ_POD_REQUESTED;
+      derivedStatuses.add(AzPodStatus.AZ_POD_REQUESTED);
     }
     logInitContainerStatuses();
     if (checkForAzPodScheduled()) {
-      return AzPodStatus.AZ_POD_SCHEDULED;
+      derivedStatuses.add(AzPodStatus.AZ_POD_SCHEDULED);
     }
     logAppContainerStatuses();
     if (checkForAzPodInitContainersRunning()) {
-      return AzPodStatus.AZ_POD_INIT_CONTAINERS_RUNNING;
+      derivedStatuses.add(AzPodStatus.AZ_POD_INIT_CONTAINERS_RUNNING);
     }
     if (checkForAzPodAppContainerStarting()) {
-      return AzPodStatus.AZ_POD_APP_CONTAINERS_STARTING;
+      derivedStatuses.add(AzPodStatus.AZ_POD_APP_CONTAINERS_STARTING);
     }
     if (checkForAzPodReady()) {
-      return AzPodStatus.AZ_POD_READY;
+      derivedStatuses.add(AzPodStatus.AZ_POD_READY);
     }
     if (checkForAzPodCompleted()) {
-      return AzPodStatus.AZ_POD_COMPLETED;
+      derivedStatuses.add(AzPodStatus.AZ_POD_COMPLETED);
     }
     if (checkForAzPodInitFailure()) {
-      return AzPodStatus.AZ_POD_INIT_FAILURE;
+      derivedStatuses.add(AzPodStatus.AZ_POD_INIT_FAILURE);
     }
     if (checkForAzPodAppFailure()) {
-      return AzPodStatus.AZ_POD_APP_FAILURE;
+      derivedStatuses.add(AzPodStatus.AZ_POD_APP_FAILURE);
+    }
+    if (derivedStatuses.size() > 1) {
+      logger.error("%d pod statuses are derived from current pod watch event",
+          derivedStatuses.size());
```

Comment:
Looks good, apart from these minor points. 
(1) Replace ther error with `log.warn`
(2) Print all the derived statuses as part of the log message, that will help with debugging future issues.

Reply:
thanks, changed to logger.warn, and logged all statuses extracted from the pod event.